dpgen2/fp/gaussian.py

```diff
@@ -0,0 +1,158 @@
+        )
+        doc_nproc = "Number of CPUs to use"
+
+        return [
+            Argument('keywords', [str, list], optional=False, doc=doc_keywords),
+            Argument('multiplicity', [int, str], optional=True, default="auto", doc=doc_multiplicity),
+            Argument('charge', int, optional=True, default=0, doc=doc_charge),
+            Argument('basis_set', str, optional=True, doc=doc_basis_set),
+            Argument('keywords_high_multiplicity', str, optional=True, doc=doc_keywords_high_multiplicity),
+            Argument('fragment_guesses', bool, optional=True, default=False, doc=doc_fragment_guesses),
+            Argument('nproc', int, optional=True, default=1, doc=doc_nproc),
+        ]
+
+    def __init__(self, **kwargs: Any):
+        self.data = kwargs
+
+
+class PrepGaussian(PrepFp):
+    def prep_task(
+            self,
+            conf_frame: dpdata.System,
+            inputs: GaussianInputs,
+    ):
+        r"""Define how one Gaussian task is prepared.
+
+        Parameters
+        ----------
+        conf_frame : dpdata.System
+            One frame of configuration in the dpdata format.
+        inputs: GaussianInputs
+            The GaussianInputs object handels all other input files of the task.
+        """
+
+        conf_frame.to('gaussian/gjf', gaussian_input_name, **inputs.data)
+
+        
+class RunGaussian(RunFp):
+    def input_files(self) -> List[str]:
+        r"""The mandatory input files to run a Gaussian task.
+        
+        Returns
+        -------
+        files: List[str]
+            A list of madatory input files names.
+
+        """
+        return [gaussian_input_name]
+
+    def optional_input_files(self) -> List[str]:
+        r"""The optional input files to run a Gaussian task.
+        
+        Returns
+        -------
+        files: List[str]
+            A list of optional input files names.
+
+        """
+        return []
+
+    def run_task(
+            self,
+            command : str,
+            out: str,
+    ) -> Tuple[str, str]:
+        r"""Defines how one FP task runs
+        
+        Parameters
+        ----------
+        command: str
+            The command of running gaussian task
+        out: str
+            The name of the output data file.
+
+        Returns
+        -------
+        out_name: str
+            The file name of the output data in the dpdata.LabeledSystem format.
+        log_name: str
+            The file name of the log.
+        """
+        # run gaussian
+        command = ' '.join([command, gaussian_input_name])
+        ret, out, err = run_command(command, shell=True)
+        if ret != 0:
+            raise TransientError(
+                'gaussian failed\n',
+                'out msg', out, '\n',
+                'err msg', err, '\n'
+            )                    
+        # convert the output to deepmd/npy format
+        sys = dpdata.LabeledSystem(gaussian_output_name, fmt='gaussian/log')
```

Comment:
Does `dpdata` assume `nopbc` for gaussian/log?

Reply:
Yes, and it also assumes `nopbc` for the input file. However, Gaussian does support the [PBC](https://gaussian.com/pbc/) condition. PBC should be implemented in dpdata.

Reply:
https://github.com/deepmodeling/dpdata/pull/397

Reply:
Great! merged.

Reply:
Fixed a bug in https://github.com/deepmodeling/dpdata/pull/401